### PR TITLE
Hidden loadout groups

### PR DESCRIPTION
**Pull request by Errant-4**
## About the PR
Makes it possible to hide a loadout group. This is useful for groups where all choices are decided by the game automatically, and adding optional choices into the group is not feasible/desirable for whatever reason. There is then no reason to display this loadout group to the player at all. (Such as survival boxes)

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Lobby/UI/Loadouts/LoadoutWindow.xaml.cs [hunk 1]
```diff
@@ -29,6 +29,9 @@ public LoadoutWindow(HumanoidCharacterProfile profile, RoleLoadout loadout, Role
             if (!protoManager.TryIndex(group, out var groupProto))
                 continue;
 
+            if (groupProto.Hidden == true)
```
Comment (deltanedas) [inline Content.Client/Lobby/UI/Loadouts/LoadoutWindow.xaml.cs:32]: remove == true its already a bool

File: Content.Shared/Preferences/Loadouts/LoadoutGroupPrototype.cs [hunk 2]
```diff
@@ -29,6 +29,12 @@ public sealed partial class LoadoutGroupPrototype : IPrototype
     [DataField]
     public int MaxLimit = 1;
 
+    /// <summary>
+    /// Hides the loadout group from the player.
+    /// </summary>
+    [DataField]
+    public bool Hidden = false;
```
Comment (deltanedas) [inline Content.Shared/Preferences/Loadouts/LoadoutGroupPrototype.cs:36]: remove = false thats the default

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Client/Lobby/UI/Loadouts/LoadoutWindow.xaml.cs (hunk 1 above)
Comment (Errant-4) [reply Content.Client/Lobby/UI/Loadouts/LoadoutWindow.xaml.cs]: Fixed

File: Content.Shared/Preferences/Loadouts/LoadoutGroupPrototype.cs (hunk 2 above)
Comment (Errant-4) [reply Content.Shared/Preferences/Loadouts/LoadoutGroupPrototype.cs]: Fixed

**Review by metalgearsloth — APPROVED** [review]
Something weird is that someone can just export and import their character to edit it anyway but I guess that's a host issue?

**Comment by Errant-4** [comment]
> Something weird is that someone can just export and import their character to edit it anyway but I guess that's a host issue?

Huh, I guess that's something to keep in mind. But in the loadout group where I wanted this, every option not blocked by species filters is force-Selected , so it's fine if the player actually presses options in it. It makes no difference

**Pull request merged**